File: 2023-1/Corte1/2-Iterativa/Raiz.java
public class Raiz {
  
  public void get_raiz(double x, double d) {
    double a = 1.0;
    while (Math.abs(a*a-x)>=d) {
      System.out.println(a);
      a = (a + x/a)/2.0;
    }

  }
  public static void main(String[] args) {
    Raiz obRaiz = new Raiz();
    obRaiz.get_raiz(49, 0.0000001);
    obRaiz.get_raiz(100,0.0000001);
  }
}

File: 2023-2/Corte1/2-Iterativa/raiz.py
"""
Carlos A Delgado
05 de Sep de 2023
Estimación de la raiz cuadrada por método de Newton
"""
def raiz(x,d):
    print(f"Raiz de {x} con tolerancia {d}")
    a = 1 #raiz inicial
    while abs(a*a-x) > d:
        a = (a+x/a)/2
        print(a, abs(a*a-x))
    return a

print(raiz(2,0.0001))
print(raiz(3,0.0001))
print(raiz(1460021,0.0001))
